### ISSUE TITLE
"make test" fails with "test/functional/helpers.lua:113: Vim(return):Error invoking 'ruby_eval' on channel 3 (ruby-script-host):'

### ISSUE THREAD

**Issue by shlomif** [neovim/neovim#15609]
### Neovim version (nvim -v)

NVIM v0.6.0-dev+229-g4eb1ebbcf

### Vim (not Nvim) behaves the same?

irrelevant

### Operating system/version

Debian x86-64 bookworm

### Terminal name/version

Konsole 21.08.1

### $TERM environment variable

screen

### Installation

build from repo

### How to reproduce the issue

```
make test
```

### Expected behavior

Tests succeed.

### Actual behavior

Tests fail with:

```
test/functional/eval/system_spec.lua:576: not tested; powershell was not found
[ SKIPPED  ] test/functional/ex_cmds/make_spec.lua @ 17: :make with powershell not tested; powershell was not found
[ SKIPPED  ] test/functional/lua/vim_spec.lua @ 1154: lua stdlib vim.opt should handle STUPID window things
[ SKIPPED  ] test/functional/normal/lang_spec.lua @ 24: gu and gU works in Turkish locale Locale tr_TR.UTF-8 not supported
[ SKIPPED  ] test/functional/normal/langmap_spec.lua @ 243: 'langmap' Translates modified keys correctly
[ SKIPPED  ] test/functional/normal/langmap_spec.lua @ 249: 'langmap' handles multi-byte characters
[ SKIPPED  ] test/functional/normal/langmap_spec.lua @ 259: 'langmap' handles multibyte mappings
[ SKIPPED  ] test/functional/options/keymap_spec.lua @ 190: 'keymap' / :lmap mappings not applied to macro replay of :lnoremap
[ SKIPPED  ] test/functional/plugin/lsp/diagnostic_spec.lua @ 860: vim.lsp.diagnostic set_signs sets signs by default
[ SKIPPED  ] test/functional/plugin/lsp_spec.lua @ 762: LSP basic_init test should check the body and didChange incremental normal mode editing
[ SKIPPED  ] test/functional/provider/nodejs_spec.lua @ 13: Missing nodejs host, or nodejs version is too old (failed to detect node)
[ SKIPPED  ] test/functional/provider/perl_spec.lua @ 17: Missing perl host, or perl version is too old ("Neovim::Ext" cpan module is not installed)
[ SKIPPED  ] test/functional/provider/python_spec.lua @ 28: Python 2 (or the pynvim module) is broken/missing (provider/pythonx: Could not load Python 2:
/usr/bin/python2 does not have the "neovim" module. :help provider-python
/usr/bin/python2.7 does not have the "neovim" module. :help provider-python
/usr/bin/python2.6 does not have the "neovim" module. :help provider-python
/usr/bin/python is Python 3.9 and cannot provide Python 2.)
[ SKIPPED  ] test/functional/terminal/tui_spec.lua @ 1373: TUI 't_Co' (terminal colors) TERM=interix uses 8 colors
[ SKIPPED  ] test/functional/ui/bufhl_spec.lua @ 475: Buffer highlighting prioritizes latest added highlight
[ SKIPPED  ] test/functional/ui/float_spec.lua @ 2134: float window with ext_multigrid supports second UI without multigrid
[ SKIPPED  ] test/functional/ui/messages_spec.lua @ 1203: ui/msg_puts_printf output multibyte characters correctly
test/functional/ui/messages_spec.lua:1221: Locale ja_JP.UTF-8 not supported
[ SKIPPED  ] test/functional/viml/lang_spec.lua @ 8: viml parses `<SID>` with turkish locale
test/functional/viml/lang_spec.lua:10: Locale tr_TR.UTF-8 not supported
[ SKIPPED  ] test/functional/viml/lang_spec.lua @ 22: viml str2float is not affected by locale
test/functional/viml/lang_spec.lua:24: Locale sv_SE.UTF-8 not supported
[  ERROR   ] 2 errors, listed below:
[  ERROR   ] test/functional/provider/ruby_spec.lua @ 115: rubyeval() evaluates ruby objects
test/functional/helpers.lua:113: Vim(return):Error invoking 'ruby_eval' on channel 3 (ruby-script-host):
Unknown request ruby_eval

stack traceback:
        test/functional/helpers.lua:113: in function 'rubyeval'
        test/functional/provider/ruby_spec.lua:116: in function <test/functional/provider/ruby_spec.lua:115>

[  ERROR   ] test/functional/provider/ruby_spec.lua @ 119: rubyeval() returns nil for empty strings
test/functional/helpers.lua:113: Vim(return):Error invoking 'ruby_eval' on channel 3 (ruby-script-host):
Unknown request ruby_eval

stack traceback:
        test/functional/helpers.lua:113: in function 'rubyeval'
        test/functional/provider/ruby_spec.lua:120: in function <test/functional/provider/ruby_spec.lua:119>


 25 SKIPPED TESTS
 2 ERRORS
-- Tests exited non-zero: 1
-- No output to stderr.
CMake Error at /home/shlomif/Download/unpack/file/vim/neovim/cmake/RunTests.cmake:86 (message):
  functional tests failed with error: 1


FAILED: CMakeFiles/functionaltest 
cd /home/shlomif/Download/unpack/file/vim/neovim/build && /usr/bin/cmake -DBUSTED_PRG=/home/shlomif/Download/unpack/file/vim/neovim/.deps/usr/bin/busted -DLUA_PRG=/home/shlomif/Download/unpack/file/vim/neovim/.deps/usr/bin/luajit -DNVIM_PRG=/home/shlomif/Download/unpack/file/vim/neovim/build/bin/nvim -DWORKING_DIR=/home/shlomif/Download/unpack/file/vim/neovim -DBUSTED_OUTPUT_TYPE=nvim -DTEST_DIR=/home/shlomif/Download/unpack/file/vim/neovim/test -DBUILD_DIR=/home/shlomif/Download/unpack/file/vim/neovim/build -DTEST_TYPE=functional -P /home/shlomif/Download/unpack/file/vim/neovim/cmake/RunTests.cmake
ninja: build stopped: subcommand failed.
make: *** [Makefile:142: functionaltest] Error 1
Error 500 Can't connect to localhost:6300 (Connection refused) at /home/shlomif/perl5/lib/perl5/App/Notifier/Client.pm line 46.
shlomif[neovim]:$trunk$ 
[0] 0:gdb  1:tmux  2:zsh* 3:zsh  4:hexchat  5:zsh-                                                                                                                                                                                                                      telaviv

```

**Comment by justinmk** [comment]
> Unknown request ruby_eval

Seems like whatever is at the other end of the pipe either

- isn't https://github.com/neovim/neovim-ruby , or...
- maybe it's really old?

in any case the tests pass on CI. Post a PR if you find an actual bug.

**Comment by seandewar** [comment]
Yeah, I believe `ruby_eval` was added to neovim-ruby in v0.9, but it's still listed as a pre-release.

I'm guessing if you run `gem list`, it displays v0.8.1 or earlier? @shlomif

**Comment by justinmk** [comment]
@alexgenco anything blocking a release?

**Comment by shlomif** [comment]
> Yeah, I believe `ruby_eval` was added to neovim-ruby in v0.9, but it's still listed as a pre-release.
> 
> I'm guessing if you run `gem list`, it displays v0.8.1 or older? @shlomif

```
[shlomif@telaviv ~]$ gem list | grep neovim
neovim (0.8.1)
[shlomif@telaviv ~]$ gem i neovim
Successfully installed neovim-0.8.1
Parsing documentation for neovim-0.8.1
Done installing documentation for neovim after 1 seconds
1 gem installed
[shlomif@telaviv ~]$ gem list | grep neovim
neovim (0.8.1)
[shlomif@telaviv ~]$ 

```

Can neovim-ruby in the ruby-gems service be fixed? alternatively, nvim's test suite can add a sanity check for its version.

**Comment by seandewar** [comment]
For now, you can try installing the pre-release version: `gem install neovim -v 0.9.0.pre.1`. @shlomif

**Comment by shlomif** [comment]
Hi!

> For now, you can try installing the pre-release version: `gem install neovim -v 0.9.0.pre.1`. @shlomif
> 
> I'm not sure if it's the _latest_ pre-release, but it should have `ruby_eval` support.

Since I've wasted a lot of time on this and wouldn't have easily thought of using gem's "--pre" flag, and per the first item in [this essay](https://www.joelonsoftware.com/2007/02/19/seven-steps-to-remarkable-customer-service/) :

Can I try adding a sanity version check for that in the test runner (e.g in https://github.com/neovim/neovim/blob/a422f2136a001921fb54e7b83b6d00e6d5f42a4a/cmake/RunTests.cmake )? I can try submitting a PR.

**Comment by shlomif** [comment]
I wrote a sanity check script in ruby for that here if there's interest (Expat licensed): https://github.com/shlomif/shlomif-computer-settings/commit/4a7c1cc738ca32192beaf427949ff416769c822d

**Comment by justinmk** [comment]
> I wrote a sanity check script in ruby for that here if there's interest (Expat licensed): [shlomif/shlomif-computer-settings@4a7c1cc](https://github.com/shlomif/shlomif-computer-settings/commit/4a7c1cc738ca32192beaf427949ff416769c822d)

Thanks. Can check the version with less code via `nvim_get_chan_info(...).client`

https://github.com/neovim/neovim-ruby/blob/b52e9265d1779df15e77673d552e9869fd4c299c/spec/acceptance/client_info_spec.vim#L12

Adding a check for the version in the tests doesn't really make sense. The point of the tests is to test that everything is working. 

To fix this issue, we need to do a release.

**Comment by alexgenco** [comment]
I'm working on a 0.9.0 release right now. I need to work through the build which has been failing for a while now

**Comment by alexgenco** [comment]
Just released [0.9.0](https://rubygems.org/gems/neovim/versions/0.9.0)
